calico_kubernetes/calico_kubernetes.py

```diff
@@ -147,6 +157,118 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _container_add(self, pid, ip, interface, hostname, orchestrator_id):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+        """
+        # Check if the container already exists. If it does, exit.
+        try:
+            _ = self._datastore_client.get_endpoint(
+                hostname=hostname,
+                orchestrator_id=orchestrator_id,
+                workload_id=self.docker_id
+            )
+        except KeyError:
+            # Calico doesn't know about this container.  Continue.
+            pass
+        else:
+            print("This container has already been configured with Calico Networking.")
+            sys.exit(1)
+
+        # Obtain information from Docker Client and validate container state
+        self._validate_container_state(self.docker_id)
+
+        # Assign ip address through IPAM Client
+        try:
+            ip_assigned = self._datastore_client.assign_address(None, ip)
+        except PoolNotFound:
+            print "ERROR: IP address %s does not belong to any configured pools."\
+                  " Exiting." % ip
+            sys.exit(1)
+        else:
+            if ip_assigned is False:
+               print "ERROR: Failed to assign IP address %s. Exiting." % ip
+
+        # Create Endpoint object
+        try:
+            ep = self._datastore_client.create_endpoint(hostname, orchestrator_id,
+                                                        self.docker_id, [ip])
+        except AddrFormatError:
+            print "This node is not configured for IPv%d. Unassigning IP "\
+                  "address %s then exiting."  % ip.version, ip
+            self._datastore_client.unassign_address(None, ip)
+            sys.exit(1)
+
+        # Create the veth, move into the container namespace, add the IP and
+        # set up the default routes.
+        ep.mac = ep.provision_veth(pid, interface)
+        self._datastore_client.set_endpoint(ep)
+
+        # Let the caller know what endpoint was created.
+        return ep
+
+    def _container_remove(self, hostname, orchestrator_id):
+        """
+        Remove the indicated container on this host from Calico networking
+        """
+        # Find the endpoint ID. We need this to find any ACL rules
+        try:
+            endpoint = self._datastore_client.get_endpoint(
+                hostname=hostname,
+                orchestrator_id=orchestrator_id,
+                workload_id=self.docker_id
+            )
+        except KeyError:
+            print("Container %s doesn't contain any endpoints" % self.docker_id)
+            sys.exit(1)
+
+        # Remove any IP address assignments that this endpoint has
+        for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
+            assert(net.size == 1)
+            ip = net.ip
```

Comment:
No need to create a whole new variable here - combine this line with the one below it.